docs/proposals/status.md

```diff
@@ -0,0 +1,300 @@
+- Propose a minimal set of standardized Conditions for all resources to implement.
+
+## Scope
+The scope is limited to the objects matching the application crd selectors. The scope is also limited to using the information present in the `.status` field for kubernetes objects. External systems to augment health (black/white box) is not part of this scope.
+
+## Proposal
+The proposal is to add a reconciler to the Application CRD controller, that creates watches for the GroupKinds that the application CRD refers to. The reconciler then inspects the matching objects and updates the aggregate status in the Application CRD. Lowering information entropy and reducing the amount of interpertation to be done by the consumers of status is important for the adoption.
+
+### Fields
+Fields are a structured way to capture the status of resources. As part of this proposal, no standardized fields are proposed. The authors opinion is it does not provide additional information than what conditions provide. If the community strongly feels about standardizing some fields for in-tree and custom resources, it can be considered. 
+
+### Conditions
+[Conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#typical-status-properties) are used to describe the status of resources in an extensible way. Out of tree controllers support a variety of conditions. There is no consistent usage across the core controller. This is exacerbated with custom controllers. To tackle this diversity and its impact in status computation for Application CRD, we need a short term and a long term solution. Short term solution is to do aggregation of diverse core resources in the Application CRD controller. Long term is to identify best practices and standardize conditions and propagate them to the core controllers thereby making Application CRD aggregation simpler.
+
+#### Standard Conditions
+These conditions should be added to `.status.conditions` for Application CRD and subsequently to other resources:
+##### `Ready`
+Readiness as perceived by the resource controller. This means the controller deems the underlying resources to be ready for consumption to the best of its knowledge. It coule be implemented as a low pass filter to filter out transient or persistent failures such as restarts of underlying pods.
+
+##### `Settled`
+Would indicate whether controller acted on latest revision of `.spec` and the underlying resources conform to the latest revision of the `.spec`
+
+##### `Error`
+Would capture Errors seen in last reconcilation loop.
+
+#### Other Conditions from core controllers
+These are not as universal for all kubernetes objects as `Ready`, `Settled`, `Error` and provide yet another dimension of `Ready` and `Settled` condition. 
+
+##### `Progressing`
+Deployment uses this to indicate there is an update to the underlying ReplicaSet that is being acted upon. This is a inversion of the `Settled` condition.
+
+##### `Available`
```

Comment:
I find `"Available"` is perpetually confusing name for this condition. People trying to build robust monitoring pipelines are nearly always caught off guard by this. It would be nice to report this as closer to what it really is -- succeeding liveness/readiness checks.